### PR TITLE
IngressController code optimization

### PR DESCRIPTION
**Pull request by ChengJie1053**
IngressController code optimization

### CODE REVIEW

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala [hunk 1]
```diff
@@ -47,15 +47,24 @@ object KubernetesRetriever extends Logger {
   // see org.apache.flink.configuration.RestOptions.RETRY_MAX_ATTEMPTS
   val FLINK_REST_RETRY_MAX_ATTEMPTS = 2
 
+  private var kubernetesClient: KubernetesClient = _
+
   /** get new KubernetesClient */
   @throws(classOf[KubernetesClientException])
-  def newK8sClient(): KubernetesClient = {
-    new DefaultKubernetesClient()
+  def getK8sClient(): KubernetesClient = {
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala:54]: There are better ways to implement the singleton pattern in scala:

```scala 
private lazy val kubernetesClient: KubernetesClient = new DefaultKubernetesClient()
def getK8sClient(): KubernetesClient = kubernetesClient
```

**Review by Al-assad — CHANGES_REQUESTED** [review]

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala [hunk 2]
```diff
@@ -107,7 +116,7 @@ object KubernetesRetriever extends Logger {
    *   deployment namespace
    */
   def isDeploymentExists(name: String, namespace: String): Boolean = {
-    using(KubernetesRetriever.newK8sClient()) {
+    using(KubernetesRetriever.getK8sClient()) {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala:119]: The KubernetesRetriever.getK8sClient() method will return a singleton KubernetesClient. However, the using method closes the KubernetesClient after used, which can lead to bugs.

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala (hunk 1 above)
Comment (ChengJie1053) [reply streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala:54]: Ok, thanks for reviewing the code

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala (hunk 2 above)
Comment (ChengJie1053) [reply streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala:119]: Ok, thanks for reviewing the code

**Pull request closed without merging**